Client/Frontend/Browser/Onboarding/OnboardingNavigationController.swift

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+
+private let log = Logger.browserLogger
+
+protocol Onboardable: class {
+    /// Show next on boarding screen if possible.
+    /// If last screen is currently presenting, the view is dimissed instead(onboarding finished).
+    func presentNextScreen(current: OnboardingViewController)
+    /// Skip all onboarding screens, onboarding is considered as completed.
+    func skip()
+}
+
+protocol OnboardingControllerDelegate: class {
+    func onboardingCompleted(_ onboardingController: OnboardingNavigationController)
+}
+
+class OnboardingNavigationController: UINavigationController {
+    
+    private struct UX {
+        /// The onboarding screens are showing as a modal on iPads.
+        static let preferredModalSize = CGSize(width: 375, height: 667)
+    }
+    
+    weak var onboardingDelegate: OnboardingControllerDelegate?
+    
+    enum Screens: CaseIterable {
+        case searchEnginePicker
+        case shieldsInfo
+        
+        func viewController(with profile: Profile) -> OnboardingViewController {
+            switch self {
+            case .searchEnginePicker:
+                return OnboardingSearchEnginesViewController(profile: profile)
+            case .shieldsInfo:
+                return OnboardingShieldsViewController(profile: profile)
+            }
+        }
+        
+        var type: AnyClass {
```

Comment:
Why not use `OnboardingViewController` instead of `AnyClass`?

Reply:
Doesn't work
```
Cannot convert return expression of type 'OnboardingSearchEnginesViewController.Type' to return type 'OnboardingViewController'
```

Reply:
It is because it is expecting an instance of `OnboardingViewController`, so need make the return type a `type`:

```suggestion
        var type: OnboardingViewController.Type {
```